# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1421861785 18000
#      Wed Jan 21 12:36:25 2015 -0500
# Node ID f3ac54ac2c6aa62acf11caf4acac987fc3f57d2b
# Parent  228cfaf24208e350cb65ce93858698e80b83c905
also skip startup message if session is not interactive

* octave.cc (octave_initialize_interpreter): If session is not
interactive or forced interactive, set inhibit_startup_message to
true.

diff --git a/libinterp/octave.cc b/libinterp/octave.cc
--- a/libinterp/octave.cc
+++ b/libinterp/octave.cc
@@ -705,20 +705,16 @@ octave_process_command_line (int argc, c
   bool script_file = (argc - optind) > 0;
 
   if (! code_to_eval.empty () && script_file)
     {
       error ("--eval \"CODE\" and script file are mutually exclusive options");
 
       octave_print_terse_usage_and_exit ();
     }
-
-  // Don't print start-up message when directly executing a script
-  if (! code_to_eval.empty () || script_file)
-    inhibit_startup_message = true;
 }
 
 // EMBEDDED is declared int instead of bool because this function is
 // declared extern "C".
 
 void
 octave_initialize_interpreter (int argc, char **argv, int embedded)
 {
@@ -793,16 +789,20 @@ octave_initialize_interpreter (int argc,
   bool stdin_is_tty = gnulib::isatty (fileno (stdin));
 
   interactive = (! embedded && stdin_is_tty
                  && gnulib::isatty (fileno (stdout)));
 
   if (! interactive && ! forced_line_editing)
     line_editing = false;
 
+  // Also skip start-up message unless session is interactive.
+  if (! (interactive || forced_interactive))
+    inhibit_startup_message = true;
+
   // Force default line editor if we don't want readline editing.
   if (! line_editing)
     command_editor::force_default_editor ();
 
   // These can come after command line args since none of them set any
   // defaults that might be changed by command line options.
 
   if (line_editing)
